src/encoder/encodermp3.cpp

```diff
@@ -134,9 +144,39 @@ void EncoderMp3::flush() {
         numBytes = lame_get_lametag_frame(
                 m_lameFlags, m_bufferOut, m_bufferOutSize);
     }
-    // Write the lame/xing header.
+
+    // Ideally, we should shift the file content forward to insert the header
+    // (ID3v2 & Xing frame) dynamically, but `EncoderCallback` doesn't support
+    // that so we use the static header padding and truncate the tracklist if
+    // too long
+    size_t tracklistMaxSize = kHeaderPadding - numBytes -
+            lame_get_id3v2_tag(m_lameFlags, nullptr, 0);
+    auto trackList = getTrackList().join("\n");
+    if (!trackList.isEmpty()) {
+        // Because of the static header size offset, we need to ensure that the
+        // tracklist comment won't make the header overflow on the MP3 frames,
+        // we we truncate to the max value
+        auto currentSize = static_cast<size_t>(trackList.size());
+        if (currentSize > tracklistMaxSize - 1) { // -1 since we need a byte for the NULL terminator
+            trackList = trackList.left(tracklistMaxSize - 4) + "...";
+        }
+        id3tag_set_comment(m_lameFlags, trackList.toLatin1());
```

Comment:
It looks like the comment is in 16 bit unicode: 
https://mutagen-specs.readthedocs.io/en/latest/id3/id3v2.2.html#id3v2-frames-overview
At least I can store any Unicode character as comment.

Reply:
ffprobe doesn't seem to support unicode

```
Input #0, mp3, from '/home/developer/Music/Mixxx/Recordings/2025-04-26_18h05m23s.mp3':
  Metadata:
    encoder         : LAME 64bits version 3.100 (http://lame.sf.net)
    comment         : 00:00:03: Odd Chap - Prohibition
                    : 00:00:31: Odd Chap - Sound 104
                    : 00:01:28: deadmau5 - Professional Griefers
                    : 00:04:36: Tristam - I Remember
                    : 00:04:45: TiÃ«sto - Dudada
                    : 00:04:55: Tristam - I Remember
                    : 00:05:13: TiÃ«sto - Dudada
  Duration: 00:05:48.38, start: 0.023021, bitrate: 128 kb/s
```

but the Gnome Player seem to render it fine (although the field doesn't support multiline...)

Reply:
Actually checked VLC and I get the opposite result than gnome player
![image](https://github.com/user-attachments/assets/07d05012-7dd5-4212-90fd-52a5c8425721)
Sadly I'm afraid there is no strong standard there, so we should probably take a stance. I think Unicode make sense IMO